disnake/utils.py

```diff
@@ -1331,6 +1332,45 @@ def assert_never(arg: Never, /) -> None:
     pass
 
 
+def signature_has_self_param(function: Callable[..., Any]) -> bool:
```

Comment:
this should also be used where we assume the parameter name is "self" in command parsing elsewhere.

Reply:
sure: https://github.com/DisnakeDev/disnake/pull/847/commits/d9184e6fef7a13c46105711c123dde66fe2d7dac (+ other commits)
I'm a bit wary of changing core slash parts of slash right now, but having tested this change a bunch, it should be fine.